source/blog/2014-02-04-ember-1-10-0-released.md

```diff
@@ -0,0 +1,387 @@
+});
+```
+
+Passing a name to the `service()` method allows a different service to be injected.
+For example:
+
+```javascript
+export default Ember.Component.extend({
+  storage: Ember.inject.service('local-storage')
+});
+```
+
+In addition to injecting services, controllers can be injected with `Ember.inject.controller`.
+
+Importantly, these injections are lazy. When used within a unit test, a stubbed service can
+be passed during `create`. For example:
+
+```javascript
+test("a value is saved on storage", function(){
+  expect(1);
+  var component = this.subject({
+    storage: {
+      write: function(){
+        ok(true, 'value was written');
+      }
+    }
+  });
+
+  var element = this.append();
+  element.click();
+});
+```
+
+Refer to the [API documentation](add this) to read about this new feature in
+detail.
+
+Thanks to [slindberg](https://github.com/slindberg) for his implementation of this feature.
+
+#### Notable Deprecations
+
+As Ember.js moves forward, various APIs are deprecated to allow for their
+removal in a later major release (such as 2.0). The
+[deprecations guide](http://emberjs.com/guides/deprecations/) summarizes
+deprecations and demonstrates how to update to a new API.
+
+* The explicit `{{bind` helper has been deprecated. This helper has
+  long been marked private, and was a legacy Sproutcore
+  feature. This helper will be removed in Ember 1.11.
+* Quote-less outlet names are deprecated in 1.10. An example of this is
+  `{{outlet modal}}`, which should be re-written as `{{outlet "modal"}}`.
+  This ensures the outlet helper is consistent with others, where unquoted
+  words are values and not string literals.
+* The `beforeObserver` feature is deprecated in Ember 1.10. Before observers
+  are rarely used, but introduce significant overhead to the observer system
+  in general. For observer use that requires the previous value of a property
+  be known, implementing a cache is simple and more efficient. Read more about
+  how to do this in [the deprecation guide](http://emberjs.com/guides/deprecations/#toc_deprecate-beforeobservers).
+* Observing the `childViews` array of a `ContainerView` is deprecated.
+* Setting the `childViews` property on a view definition is deprecated in
+  1.10. For example:
+
+```js
+var ChildB = Ember.View.extend();
+
+export default Ember.ContainerView.extend({
+  childViews: [ 'childA', ChildB ],
+  childA: Ember.View.extend()
+});
+```
+
+This use of `childViews` is inconsistent with other uses throughout Ember, and
+as a result is difficult to implement with good performance. Explicitly creating
+views upon initialization is preferred:
+
+```js
+var ChildB = Ember.View.extend();
+
+export default Ember.ContainerView.extend({
+  init: function(){
+    this._super();
+    this.pushObject(this.createChildView(this.childA));
+    this.pushObject(this.createChildView(ChildB));
+  },
+  childA: Ember.View.extend()
+});
+```
+
+## New Features in Ember.js 1.11
```

Comment:
Maybe say "1.11 beta" and start the following paragraph with "We're happy to announce the availability of the first beta in the 1.11 series."